File: week_1/day-5-string-hashing/variations_que/longestPalindromeLength.js
function longestPalindrome(s) {
    const freq = {};
    for (let ch of s) freq[ch] = (freq[ch] || 0) + 1;

    let len = 0;
    let odd = false;

    for (let k in freq) {
        len += Math.floor(freq[k] / 2) * 2;
        if (freq[k] % 2 === 1) odd = true;
    }
    return odd ? len + 1 : len;
}

console.log(longestPalindrome("abccccdd")); // 7

File: week_1/day-5-string-hashing/variations_que/longest_palindrome_length.py
def longestPalindrome(s):
    freq = {}
    for ch in s:
        freq[ch] = freq.get(ch, 0) + 1

    length = 0
    odd_found = False

    for count in freq.values():
        length += (count // 2) * 2
        if count % 2 == 1:
            odd_found = True

    return length + 1 if odd_found else length

# INPUT
print(longestPalindrome("abccccdd"))  # 7
